functions/go/create-setters/createsetters/create_setters_test.go

```diff
@@ -0,0 +1,304 @@
+  env: [foo, bar]
+ `,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment # kpt-set: ${name}-deployment
+  env: [foo, bar] # kpt-set: ${env}
+`,
+		},
+		{
+			name: "set comment for scalar nodes",
+			config: `
+data:
+  name: nginx
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  env: [foo, bar]
+ `,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment # kpt-set: ${name}-deployment
+  env: [foo, bar]
+`,
+		},
+		{
+			name: "set comment for scalar and sequence nodes",
+			input: `apiVersion: v1
+kind: Service
+metadata:
+  name: myService
+  namespace: foo 
+image: nginx:1.7.1 
+env: [foo, bar] 
+`,
+			config: `
+data:
+  app: myService
+  ns: foo
+  image: nginx
+  tag: 1.7.1
+  env: "[foo, bar]"
+`,
+			expectedResources: `apiVersion: v1
+kind: Service
+metadata:
+  name: myService # kpt-set: ${app}
+  namespace: foo # kpt-set: ${ns}
+image: nginx:1.7.1 # kpt-set: ${image}:${tag}
+env: # kpt-set: ${env}
+  - foo # kpt-set: ${ns}
+  - bar
+`,
+		},
+
+		{
+			name: "scalar setter donot match",
+			config: `
+data:
+  name: ubuntu
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  env: [foo, bar]
+ `,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  env: [foo, bar]
+`,
+		},
+		{
+			name: "array setter with flow style donot match",
+			config: `
+data:
+  env: |
+    [foo, bar, pro]
+  name: nginx
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  env: [foo, bar]
+ `,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment # kpt-set: ${name}-deployment
+  env: [foo, bar]
+`,
+		},
+		{
+			name: "create array setter with scalar error",
+			config: `
+data:
+  app: myService
+  ns: foo
+  images: |
+    - ubuntu
+    - linux
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  images:
+    - nginx
+    - ubuntu
+`,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  images:
+    - nginx
+    - ubuntu
+`,
+		},
+		{
+			name: "containing overlap values",
+			config: `
+data:
+  image: nginx
+  name: image
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+`,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment # kpt-set: ${image}-deployment
+`,
+		},
+		{
+			name: "FlowStyle to FoldedStyle",
+			config: `
+data:
+  image: "[nginx, ubuntu]"
+  os: ubuntu
+`,
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: [nginx, ubuntu]
+`,
+			expectedResources: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: # kpt-set: ${image}
+    - nginx
+    - ubuntu # kpt-set: ${os}
+`,
+		},
+		{
+			name: "Empty array values",
```

Comment:
This is allowed. We should add comment to sequence node with empty values.

name: [] # kpt-set: ${image}

Reply:
```
data:  
  image: "[]"
```

```
apiVersion: apps/v1
kind: Deployment
metadata:  
  name: []
```

expected:
```
apiVersion: apps/v1
kind: Deployment
metadata:  
  name: [] # kpt-set: ${image}
```

Reply:
For empty arrays, leave them as flow style. Refer to apply-setters code.